aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,167 @@
+
+Usually, any response not natively compressed can benefit from response compression. Responses not natively compressed typically include: CSS, JavaScript, HTML, XML, and JSON. You shouldn't compress natively compressed assets, such as PNG files, which are already compressed. If you attempt to further compress a natively compressed response, any small additional reduction in size and transmission time will likely be overshadowed by the time it took to process the compression.
+
+When a client can process compressed content, the client must inform the server of its capabilities by sending the `Accept-Encoding` header with the request. When a server sends compressed content, it must include information in the `Content-Encoding` header on how the compressed response is encoded. Content encoding designations are shown below indicating which ones are supported by the middleware.
+
+Content Encoding | Supported | Description
+:---: | :---: | ---
+`br` |  No | Brotli Compressed Data Format
+`compress` | No | UNIX "compress" data format
+custom | Yes | Developer provides the compression implementation. The client must be able to decompress the payload.
+`deflate` |  No | "deflate" compressed data inside the "zlib" data format 
+`exi` | No | W3C Efficient XML Interchange
+`gzip` | Yes (default) | GZip file format
+`identity` | Yes | "No encoding" identifier: The response must not be encoded.
+`pack200-gzip` | No | Network Transfer Format for Java Archives
+`*` | Yes | Any available content encoding not explicitly requested
+
+For more information, see the [IANA Official Content Coding List](http://www.iana.org/assignments/http-parameters/http-parameters.xml#http-content-coding-registry).
+
+The middleware is capable of reacting to quality value (qvalue, `q`) weighting when sent by the client. For more information, see [RFC 7231: Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4).
+
+Most compression algorithms are subject to a tradeoff between compression speed and the effectiveness of the compression. *Effectiveness* in this context refers to the size of the output after compression. The smallest size is achieved by the most *optimal* compression. The middleware defaults to the fastest compression level, which might not produce the most efficient compression. If the most efficient compression is desired, the middleware can be configured for optimal compression.
```

Comment:
The sample illustrates how to set it but I guess it can be configured to something other than the default value.